Source/WebCore/loader/LocalFrameLoaderClient.h

```diff
@@ -103,7 +103,7 @@ enum class WasPrivateRelayed : bool;
 
 struct StringWithDirection;
 
-class WEBCORE_EXPORT LocalFrameLoaderClient : public FrameLoaderClient {
+class WEBCORE_EXPORT LocalFrameLoaderClient : public FrameLoaderClient, public RefCounted<LocalFrameLoaderClient> {
```

Comment:
My alternative proposal:
1. Don't subclass RefCounted here
2. Have LocalFrameLoaderClient forward its ref-counting to FrameLoader
3. Have FrameLoader forward its ref counting to LocalFrame

Reply:
@rniwa @geoffreygaren What do you think?

Reply:
So FrameLoader isn't ref counted either so it seems that we need to make it also forward ref to LocalFrame?

Reply:
Yes, this is step 3 of my plan above.

Reply:
Starting: https://github.com/WebKit/WebKit/pull/34656

Reply:
Alternative in https://github.com/WebKit/WebKit/pull/34705

Reply:
Closing this now https://github.com/WebKit/WebKit/pull/34705 landed.